app/charts/chart-loading-wrapper.tsx

```diff
@@ -74,6 +75,8 @@ export const ChartLoadingWrapper = <
 
   if (metadata && dimensions && measures && observations) {
     const { title } = metadata;
+    const dimensionsByIri = keyBy(dimensions, (d) => d.iri);
+    const measuresByIri = keyBy(measures, (d) => d.iri);
```

Comment:
Nice 💯 

Nit: Maybe you should memo the two variables so that props of the created chart below are stable. That would enable the use of React.memo on the charts.